// access-sniff/dist/logger/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.log = exports.generalError = exports.errorMessage = exports.lintFree = exports.finishedMessage = exports.startMessage = exports.generalMessage = exports.default = undefined;

var _generalMessage = require('./generalMessage');

var _generalMessage2 = _interopRequireDefault(_generalMessage);

var _startMessage = require('./startMessage');

var _startMessage2 = _interopRequireDefault(_startMessage);

var _finishedMessage = require('./finishedMessage');

var _finishedMessage2 = _interopRequireDefault(_finishedMessage);

var _lintMessage = require('./lintMessage');

var _lintMessage2 = _interopRequireDefault(_lintMessage);

var _errorMessage = require('./errorMessage');

var _errorMessage2 = _interopRequireDefault(_errorMessage);

var _generalErrorMessage = require('./generalErrorMessage');

var _generalErrorMessage2 = _interopRequireDefault(_generalErrorMessage);

var _logMessage = require('./logMessage');

var _logMessage2 = _interopRequireDefault(_logMessage);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var logger = { generalMessage: _generalMessage2.default, startMessage: _startMessage2.default, finishedMessage: _finishedMessage2.default, lintFree: _lintMessage2.default, errorMessage: _errorMessage2.default, generalError: _generalErrorMessage2.default, log: _logMessage2.default };

exports.default = logger;
exports.generalMessage = _generalMessage2.default;
exports.startMessage = _startMessage2.default;
exports.finishedMessage = _finishedMessage2.default;
exports.lintFree = _lintMessage2.default;
exports.errorMessage = _errorMessage2.default;
exports.generalError = _generalErrorMessage2.default;
exports.log = _logMessage2.default;
