// access-sniff/dist/messages/buildMessage.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ignoredCheck = exports.default = undefined;

var _underscore = require('underscore');

var _underscore2 = _interopRequireDefault(_underscore);

var _getElementPosition = require('../helpers/getElementPosition');

var _getElementPosition2 = _interopRequireDefault(_getElementPosition);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ignoredCheck = function ignoredCheck(ignoredRules, error) {
  if (!error) {
    return false;
  }

  return _underscore2.default.some(ignoredRules, function (rule) {
    return error.startsWith(rule);
  });
};

var buildMessage = function buildMessage(msg, fileContents, _ref) {
  var ignore = _ref.ignore,
      reportLevelsArray = _ref.reportLevelsArray;

  var msgSplit = msg.split('|');
  var message = void 0;

  var ignored = ignoredCheck(ignore, msgSplit[1]);

  if (ignored) {
    return message;
  }

  // Start the Logging if the the report level matches
  if (_underscore2.default.contains(reportLevelsArray, msgSplit[0])) {
    message = {
      heading: msgSplit[0],
      issue: msgSplit[1],
      description: msgSplit[2],
      position: (0, _getElementPosition2.default)(msgSplit[3], fileContents),
      element: {
        node: msgSplit[3],
        class: msgSplit[4],
        id: msgSplit[5]
      }
    };
  }

  // Return the message for reports
  return message;
};

exports.default = buildMessage;
exports.ignoredCheck = ignoredCheck;
